### PR TITLE
fix validation: Only set tool_choice `auto` if at least one tool is provided

### PR DESCRIPTION
**Pull request by chiragjn**
Passing 

```
{
...
"tools": null
}
```

without "tool_choice" key causes a validation failure because the code sets "tool_choice" to "auto" on its own and then fails because it finds "tools" as None.
This change fixes the validation so that "tool_choice" is only set if at least one tool is provided

### CODE REVIEW

File: vllm/entrypoints/openai/protocol.py [hunk 1]
```diff
@@ -381,11 +381,12 @@ def check_tool_usage(cls, data):
...
         # default to "auto" tool_choice
-        if "tool_choice" not in data and "tools" in data:
-            data["tool_choice"] = "auto"
+        if "tool_choice" not in data:
+            if data.get("tools"):
+                data["tool_choice"] = "auto"
 
         # if "tool_choice" is specified -- validation
-        if "tool_choice" in data:
+        elif "tool_choice" in data:
```
Comment (DarkLight1337) [inline vllm/entrypoints/openai/protocol.py:389]: Why change this to `elif`?

Comment (chiragjn) [reply vllm/entrypoints/openai/protocol.py:389]: Should be mutually exclusive with line 384, right?

But I think post the `if data.get("tools"):` change on line 385, it should be fine to keep it `if`
The reason I made it elif because otherwise setting tool_choice on line 386 will also trigger this branch

Comment (DarkLight1337) [reply vllm/entrypoints/openai/protocol.py:389]: They are not mutually exclusive. In the case of `"tool_choice" not in data and data.get("tools")`, it's intended to run the code inside the subsequent `if` block.

Comment (DarkLight1337) [reply vllm/entrypoints/openai/protocol.py:389]: I think to fix this bug, the following code should be sufficient:

```diff,py
-        if "tool_choice" not in data and "tools" in data:
+        if "tool_choice" not in data and data.get("tools"):
             data["tool_choice"] = "auto"
 
         # if "tool_choice" is specified -- validation
         if "tool_choice" in data:
```

Comment (chiragjn) [reply vllm/entrypoints/openai/protocol.py:389]: Sure, applying your suggestion then

**Comment by DarkLight1337** [comment]
Can you add a test to check the behavior when `"tools": null` is passed? Otherwise looks good.

**Comment by chiragjn** [comment]
I have added some basic tests

**Review by DarkLight1337 — APPROVED** [review]
Thanks for the bugfix!

**Review by K-Mistele — APPROVED** [review]
LGTM :)
